mongo-stream.js

```diff
@@ -0,0 +1,252 @@
+const BSON = require('bson');
+const bson = new BSON();
+const elasticsearch = require('elasticsearch');
+const fs = require('fs');
+const MongoClient = require('mongodb').MongoClient;
+
+
+class MongoStream {
+  constructor(esClient, db, limit) {
+    this.esClient = esClient;
+    this.db = db;
+    this.changeStreams = {};
+    this.dumpLimit = limit;
+  }
+
+  // constructs and returns a new MongoStream
+  static async init(options) {
+    const client = await MongoClient.connect(options.url, options.mongoOpts);
+    const db = client.db(options.db);
+    const esClient = new elasticsearch.Client(options.elasticOpts);
+    const mongoStream = new MongoStream(esClient, db, options.dumpLimit);
+
+    let filterFunction;
+    if (options.coll_inclusive) {
+      filterFunction = function (collectionName) {
+        return options.coll_inclusive.indexOf(collectionName) > -1;
+      }
+    }
+    else if (options.coll_exclusive) {
+      filterFunction = function (collectionName) {
+        return options.coll_exclusive.indexOf(collectionName) === -1;
+      }
+    }
+    else {
+      filterFunction = function (collectionName) {
+        return true;
+      }
+    }
+
+    const collections = await db.collections();
+    for (let i = 0; i < collections.length; i++) {
+      const collectionName = collections[i].collectionName;
+      if (!filterFunction(collectionName)) continue;
+      mongoStream.addChangeStream(collectionName);
+    }
+
+    return mongoStream;
+  }
+
+  // delete all docs in ES before dumping the new docs into it
+  async deleteESCollection(collectionName) {
+    let searchResponse;
+    try {
+      // First get a count for all ES docs of the specified type
+      searchResponse = await this.esClient.search({
+        index: this.db.databaseName,
```

Comment:
Probably should be moved to the ES options. This assumption might not be true for everyone